src/renderer/services/chain/fees/swap.ts

```diff
@@ -0,0 +1,62 @@
+import * as RD from '@devexperts/remote-data-ts'
+import { Fees } from '@xchainjs/xchain-client'
+import { Chain, Asset, BNBChain, THORChain, BTCChain, ETHChain, baseAmount, BaseAmount } from '@xchainjs/xchain-util'
+import * as FP from 'fp-ts/lib/function'
+import * as Rx from 'rxjs'
+
+import { getChainAsset } from '../../../helpers/chainHelper'
+import { LiveData, liveData } from '../../../helpers/rx/liveData'
+import * as BNB from '../../binance'
+import * as BTC from '../../bitcoin'
+import * as THOR from '../../thorchain'
+import { FeesLD, SwapFeesLDS } from '../types'
+
+const reloadSwapFees = () => {
+  BNB.reloadFees()
+  BTC.reloadFees()
+  THOR.reloadFees()
+}
+
+/**
+ * @todo rethink about using common fees at all views
+ * and move view-related calculations directly to the views
+ * instead of creating fee-services for every situations
+ */
+const feesByChain$ = (chain: Chain): FeesLD => {
+  switch (chain) {
+    case BNBChain:
+      return BNB.fees$
+
+    case THORChain:
+      return THOR.fees$
+
+    case BTCChain:
+      return FP.pipe(
+        BTC.fees$,
+        liveData.map((btcFees) => btcFees.fees)
```

Comment:
`BTC.fees$` is wrong! Check[ its documentation](https://github.com/thorchain/asgardex-electron/blob/20b3098c39748f4b96435856339fa652ee555d56/src/renderer/services/bitcoin/fees.ts#L35): `Transaction fees (no memo included)`. In case of `source` fee, a memo has to be passed to [`poolFee$`](https://github.com/thorchain/asgardex-electron/blob/20b3098c39748f4b96435856339fa652ee555d56/src/renderer/services/bitcoin/fees.ts#L71). Again, pls check implementation of `Deposit` or `Withdraw`!

Reply:
does it make sense to do it here ? previously i've checked swap's memo format and there is no amount information there. will this memo affect anything ?

Reply:
If we send any memo with BTC, it will be added to OP_RETURN field of BTC tx. That's why fees will be calculated in a different way, check out [`calcFee` in xchain-bitcoin](https://github.com/xchainjs/xchainjs-lib/blob/ad73f5746196a55ff6a5967fe5f7af7524a5832d/packages/xchain-bitcoin/src/client.ts#L222-L229) to see the difference.

Reply:
It seems you have already implemented it with 74175d0, right?

Reply:
@veado yes. for some reason i forgot to push that changes oO